src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -92,6 +95,18 @@ void PosixEventEngine::Run(EventEngine::Closure* closure) {
   executor_.Run(closure);
 }
 
+absl::Status PosixEventEngine::WaitForPendingTasks(Duration timeout) {
+  auto deadline = ToTimestamp(timer_manager_.Now(), timeout);
+  while (executor_.IsBusy()) {
+    if (timer_manager_.Now() > deadline) {
+      return absl::DeadlineExceededError(
+          "Timed out waiting for executor to become idle.");
+    }
+    absl::SleepFor(absl::Milliseconds(33));
```

Comment:
How was this constant chosen?  Maybe document?

Reply:
Totally arbitrary, and I suspect 30 checks per second is overkill for this method. I considered using exponential backoff from say 20 milliseconds up to 1/4 the timeout (again arbitrary numbers). We don't have real workloads yet, so it's hard to tune. Do you have a recommendation?